app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/MainActivity.java

```diff
@@ -0,0 +1,220 @@
+package nyc.c4q.jonathancolon.inContaq.contactlist.activities;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.ContentResolver;
+import android.content.pm.PackageManager;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.ContactsContract;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.io.BufferedInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import nyc.c4q.jonathancolon.inContaq.R;
+import nyc.c4q.jonathancolon.inContaq.contactlist.Contact;
+import nyc.c4q.jonathancolon.inContaq.utlities.sqlite.ContactDatabaseHelper;
+
+import static android.provider.ContactsContract.CommonDataKinds.Email;
+import static android.provider.ContactsContract.CommonDataKinds.Phone;
+import static android.provider.ContactsContract.Contacts;
+import static nl.qbusict.cupboard.CupboardFactory.cupboard;
+
+public class MainActivity extends Activity {
+    public TextView outputText;
+    private SQLiteDatabase db;
+
+
+    private final String[] permissions = new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE,
+            Manifest.permission.READ_SMS, Manifest.permission.READ_CONTACTS};
+    private ImageView profileImage;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_import_contacts);
+        outputText = (TextView) findViewById(R.id.textView1);
+        profileImage = (ImageView) findViewById(R.id.profile_image);
+        checkPermissions();
+        fetchContacts();
+    }
+
+    synchronized public void fetchContacts() {
+
+        String phoneNumber = null;
+        String email = null;
+
+        Uri CONTENT_URI = Contacts.CONTENT_URI;
+        String _ID = Contacts._ID;
+        String DISPLAY_NAME = Contacts.DISPLAY_NAME;
+        String HAS_PHONE_NUMBER = Contacts.HAS_PHONE_NUMBER;
+
+        Uri PhoneCONTENT_URI = Phone.CONTENT_URI;
+        String Phone_CONTACT_ID = Phone.CONTACT_ID;
+        String NUMBER = Phone.NUMBER;
+        String TYPE = Phone.TYPE;
+
+        Uri EmailCONTENT_URI = Email.CONTENT_URI;
+        String EmailCONTACT_ID = Email.CONTACT_ID;
+        String DATA = Email.DATA;
+
+        StringBuffer output = new StringBuffer();
+        ContentResolver contentResolver = getContentResolver();
+        Cursor cursor = contentResolver.query(CONTENT_URI, null, null, null, null);
+
+        // Loop for every contact in the phone
+        if (cursor.getCount() > 0) {
```

Comment:
The convention (iirc) is cursor.moveToFirst(), which returns false if the cursor is empty